pkg/controllers/nodeclass/status/nodeimage.go

```diff
@@ -0,0 +1,179 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/logging"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily"
+	"github.com/awslabs/operatorpkg/reasonable"
+	"github.com/samber/lo"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/utils/pretty"
+)
+
+type NodeImageReconciler struct {
+	nodeImageProvider imagefamily.NodeImageProvider
+	cm                *pretty.ChangeMonitor
+}
+
+func NewNodeImageReconciler(provider imagefamily.NodeImageProvider) *NodeImageReconciler {
+	return &NodeImageReconciler{
+		nodeImageProvider: provider,
+		cm:                pretty.NewChangeMonitor(),
+	}
+}
+
+func (r *NodeImageReconciler) Register(_ context.Context, m manager.Manager) error {
+	return controllerruntime.NewControllerManagedBy(m).
+		Named("nodeclass.nodeimage").
+		For(&v1alpha2.AKSNodeClass{}).
+		WithOptions(controller.Options{
+			RateLimiter:             reasonable.RateLimiter(),
+			MaxConcurrentReconciles: 10,
```

Comment:
Same comment as other, though not necessarily expecting you to solve it here.

Reply:
Added same TODO tomment to all status Reconcilers.